action/protocol/staking/read_state.go

```diff
@@ -216,3 +231,7 @@ func getTotalStakedAmount(ctx context.Context, csr CandidateStateReader) (*big.I
 	// otherwise read from bucket pool
 	return csr.TotalStakedAmount(), nil
 }
+
+func getActiveBucketsCount(csr CandidateStateReader) uint64 {
```

Comment:
i mean, similar to the code in getTotalStakedAmount()

if after greenland {
     // read state from db
    return total.count
}

return csr.ActiveBucketsCount(), nil